apps/server/src/shared/domain/rules/team.rule.spec.ts

```diff
@@ -33,35 +30,65 @@ describe('TeamRule', () => {
 		await orm.close();
 	});
 
-	beforeEach(() => {
-		role = roleFactory.build({ permissions: [permissionA] });
-		teamRole = roleFactory.build({ permissions: [permissionB] });
-		user = userFactory.build({ roles: [role] });
-		entity = teamFactory.withRoleAndUserId(teamRole, user.id).build();
-	});
+	const setup = () => {
+		const role = roleFactory.buildWithId({ permissions: [permissionA] });
+		const teamRole1 = roleFactory.buildWithId({ permissions: [teamPermissionA] });
+		const teamRole2 = roleFactory.buildWithId({ permissions: [teamPermissionB], roles: [teamRole1] });
+		const teamRole = roleFactory.buildWithId({ permissions: [teamPermissionC], roles: [teamRole2] });
+		const user = userFactory.buildWithId({ roles: [role] });
+		const entity = teamFactory.withRoleAndUserId(teamRole, user.id).build();
+		return {
+			role,
+			teamRole1,
+			teamRole2,
+			teamRole,
+			user,
+			entity,
+		};
+	};
 
 	describe('isApplicable', () => {
 		it('should return truthy', () => {
-			expect(() => service.isApplicable(entity.teamUsers[0].user, entity)).toBeTruthy();
+			const { user, entity } = setup();
+			expect(() => service.isApplicable(user, entity)).toBeTruthy();
 		});
 	});
 
 	describe('hasPermission', () => {
-		it('should return "true" if user in scope', () => {
-			const res = service.hasPermission(entity.teamUsers[0].user, entity, PermissionContextBuilder.read([permissionB]));
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionA]));
 			expect(res).toBe(true);
 		});
 
-		it('should return "false" if user has not permission', () => {
-			const res = service.hasPermission(entity.teamUsers[0].user, entity, PermissionContextBuilder.read([permissionC]));
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionB]));
+			expect(res).toBe(true);
+		});
+
+		it('should return "true" if user in team scope', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionC]));
+			expect(res).toBe(true);
+		});
+
+		it('should return "false" if user in scope but without permission', () => {
+			const { user, entity } = setup();
+			const res = service.hasPermission(user, entity, PermissionContextBuilder.read([teamPermissionD]));
+			expect(res).toBe(false);
+		});
+
+		it('should return "false" if user has global permission', () => {
```

Comment:
I'm really struggling to understand the test cases here. I think it's mainly because of the implementation of the teams, so not part of this PR. But from what I get: It is completely irrelevant which roles the actual user has. Correct? I guess this is what you want to test here, right? Maybe one could name the test 'should return false if required permission is not contained in the teamUser's permissions even if it is contained in the actual user's permissions'. The test below could be deleted then as it is contained in this one.